CMakeLists.txt

```diff
@@ -1,16 +1,27 @@
 cmake_minimum_required(VERSION 3.5)
 project(LibreCuda CXX)
 
-# create elf-parser library
-add_subdirectory(third_party/ELFIO)
+set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
+set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
 
-add_executable(LibreCuda
-        src/main.cpp
-        src/librecuda.cpp
-        src/librecuda_status.cpp
-        src/cmdqueue.cpp)
 
-target_link_libraries(LibreCuda PRIVATE elfio)
+if(NOT EXISTS ${CMAKE_SOURCE_DIR}/third_party/ELFIO/elfio)
+        message("-- ELFIO not found, fetching ELFIO...")
+        execute_process(COMMAND git submodule update --init)
+else ()
+        message("-- ELFIO found.")
+        add_subdirectory(third_party/ELFIO)
+endif()
 
-# TODO: REMOVE WHEN CRUDE DEBUG MAIN GETS DELETED
-file(COPY ${CMAKE_SOURCE_DIR}/write_float.cubin DESTINATION ${CMAKE_BINARY_DIR})
\ No newline at end of file
+set(
+    DRIVERAPI_SOURCES
+    ${CMAKE_SOURCE_DIR}/src/librecuda.cpp
+    ${CMAKE_SOURCE_DIR}/src/librecuda_status.cpp
+    ${CMAKE_SOURCE_DIR}/src/cmdqueue.cpp
+)
+
+add_library(driverapi SHARED ${DRIVERAPI_SOURCES})
+target_include_directories(driverapi PRIVATE ${CMAKE_SOURCE_DIR}/include)
+target_link_libraries(driverapi PRIVATE elfio)
```

Comment:
Use PUBLIC here for include/, and no user of driverapi will ever have to wory again about including its headers